### PR TITLE
add cache clearing to apollo example

### PR DESCRIPTION
**Pull request by puglyfe**
## Changes

- Adds cache clearing to `examples/graphql-react-apollo`

This gotcha is called out in the [MSW FAQs](https://mswjs.io/docs/faq#apollo-client) but isn't specifically addressed in the example. This is likely to come up for any non-trivial implementation, so I think it would be helpful to include.

### CODE REVIEW

File: examples/graphql-react-apollo/src/setupTests.js [hunk 1]
```diff
@@ -3,13 +3,20 @@
...
 
 beforeAll(() => {
   // Enable the mocking in tests.
   server.listen()
 })
 
+beforeEach(() => {
+  // Ensure Apollo cache is cleared between tests.
+  // https://www.apollographql.com/docs/react/api/core/ApolloClient/#ApolloClient.clearStore
+  return client.clearStore()
```
Comment (kettanaito) [inline examples/graphql-react-apollo/src/setupTests.js:17]: What is the difference between `client.clearStore()` and `client.cache.reset()`?

I've been always using the latter one to ensure there's no cache between tests in Apollo. It also looks like a more semantic thing to do based on the methods called.

Comment (puglyfe) [reply examples/graphql-react-apollo/src/setupTests.js:17]: `clearStore()` resets the QueryManager and cancels any pending queries before clearing the cache (it still calls `cache.reset()` under the hood) https://github.com/apollographql/apollo-client/blob/main/src/core/QueryManager.ts#L698-L724

It seems to be the blessed path for resetting the cache, as `InMemoryCache.reset()` isn't mentioned in the class docs https://www.apollographql.com/docs/react/api/cache/InMemoryCache/

In the case of testing, either is likely to work, so if you'd rather I switch it to `client.cache.reset()` I can do that.

Comment (kettanaito) [reply examples/graphql-react-apollo/src/setupTests.js:17]: Request cancellation sounds great in the `beforeEach` hook! Thanks for explaining this.

**Review by kettanaito — APPROVED** [review]

**Comment by kettanaito** [comment]
Welcome to contributors, @puglyfe! 🎉

**Pull request merged**